build.gradle

```diff
@@ -83,8 +84,7 @@ dependencies {
         exclude group: 'org.eclipse.jgit', module: 'org.eclipse.jgit'
     }
 
-    testImplementation group: 'org.testcontainers', name: 'spock', version: '1.15.3'
-    testImplementation group: 'org.codehaus.groovy', name: 'groovy-all', version: '3.0.7'
+    testImplementation group: 'org.testcontainers', name: 'spock', version: '1.16.0'
```

Comment:
I'm not sure if I follow.
Please take a look at the following test: https://github.com/allegro/axion-release-plugin/blob/b6e82a72d85293f57fd7141bbc2a07007291bb32/src/integration/groovy/pl/allegro/tech/build/axion/release/RemoteRejectionTest.groovy

What it does:
1. builds a custom Docker image from a `Dockerfile`
2. runs it
3. checks code behavior against a real server

If there are users without Docker I can pass a flag to this test that will ignore it if Docker is not present at host machine. Seems a good trade-off for me (test will be run on CI)

Reply:
for integrations tests it's ok, but when you remove groovy-all+spck dependency you won't be able to run unit tests without docker, right?

Reply:
`groovy-all` is provided by Gradle itself, I added explicit declaration here: https://github.com/allegro/axion-release-plugin/blame/24cecc613af9cca43305c8f3265c28e2de3c4bc6/build.gradle#L88 
but the explicit declaration of groovy is error-prone. Consider the following future situation:

1. You update Gradle to a newer version, 
2. Gradle comes with a new Groovy version
3. Whoops - you can't build axion because of groovy version conflicts 

That's why it's better to have implicit groovy dependency instead of an explicit one.

So to summarize changes in lines `86-89`: 
I bumped [tesctontainers](https://www.testcontainers.org/test_framework_integration/spock/) library (used by `RemoteRejectionTest.groovy`)
I removed the explicit groovy dependency declaration.

Reply:
ok, i mixed spock-core and test-containers spock
groovy-all from gradle seems ok, since this is a plugin